app.py

```diff
@@ -9,43 +11,80 @@
 parser.add_argument('name')
 parser.add_argument('function')
 
+params_schema = Schema({'name':str,'function':str})
+id_schema = Schema(All(str,Length(min=24,max=24,),Match(r'^[0123456789abcdef]*$')))
 
 class Thing(Resource):
-    def get(self, name):
-        thing = Model.get_by_name(name)
+    def get(self, _id):
+        try:
+            id_schema(_id)
+        except:
+            abort(400,message='invalid id')
```

Comment:
do all of this within one try / except block ... the except will just except multiple errors thrown and handle them in their own ways